### PR TITLE
support param masters in cce cluster

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
Test result:
make testacc TEST='./huaweicloud' TESTARGS='-run TestAccCCEClusterV3_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run TestAccCCEClusterV3_basic -timeout 360m -parallel 4
=== RUN   TestAccCCEClusterV3_basic
=== PAUSE TestAccCCEClusterV3_basic
=== CONT  TestAccCCEClusterV3_basic
--- PASS: TestAccCCEClusterV3_basic (422.07s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       422.138s

### CODE REVIEW

File: docs/resources/cce_cluster.md [hunk 1]
```diff
@@ -145,6 +145,12 @@ versions are available, choose Dashboard > Buy Cluster on the CCE console. Chang
 
 * `enterprise_project_id` - (Optional, String, ForceNew) The enterprise project id of the cce cluster. Changing this creates a new cluster.
 
+* `masters` - (Optional, List, ForceNew) Advanced configuration of master nodes. Changing this creates a new cluster. Conflicts with `multi_az`.
```
Comment (ShiChangkuo) [inline docs/resources/cce_cluster.md]: seems `masters` and `multi_az` have the same logical function, so please put them together.

File: docs/resources/cce_cluster.md [hunk 2]
```diff
@@ -134,7 +134,7 @@ versions are available, choose Dashboard > Buy Cluster on the CCE console. Chang
 * `authenticating_proxy_ca` - (Optional, String, ForceNew) CA root certificate provided in the authenticating_proxy mode. The CA root certificate
 	is encoded to the Base64 format. Changing this parameter will create a new cluster resource.
 
-* `multi_az` - (Optional, Bool, ForceNew) Enable multiple AZs for the cluster, only when using HA flavors. Changing this parameter will create a new cluster resource.
+* `multi_az` - (Optional, Bool, ForceNew) Enable multiple AZs for the cluster, only when using HA flavors. Changing this parameter will create a new cluster resource. Conflicts with `masters`.
```
Comment (ShiChangkuo) [inline docs/resources/cce_cluster.md]: two small tips:
1. if the paramter conflicts with others, please use the following template:
*This parameter and `xxx` are alternative.*
2. In code style guides, maximum line length rules are often set to 100 or 120.

File: huaweicloud/resource_huaweicloud_cce_cluster_v3.go [hunk 3]
```diff
@@ -145,6 +147,24 @@ func ResourceCCEClusterV3() *schema.Resource {
 				ForceNew: true,
 				Computed: true,
 			},
+			"masters": {
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_cce_cluster_v3.go:133]: ditto.

File: huaweicloud/resource_huaweicloud_cce_cluster_v3.go [hunk 4]
```diff
@@ -238,6 +258,25 @@ func resourceClusterExtendParamV3(d *schema.ResourceData, config *Config) map[st
 	return m
 }
 
+func resourceClusterMastersV3(d *schema.ResourceData) ([]clusters.MasterSpec, error) {
+	flavorId := d.Get("flavor_id").(string)
+	mastersRaw := d.Get("masters").([]interface{})
+	if strings.Split(flavorId, ".")[1] == "s1" && len(mastersRaw) != 1 {
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_cce_cluster_v3.go]: seems the flavor likes `cce.dec.s1.small` does not applicable.

File: huaweicloud/resource_huaweicloud_cce_cluster_v3.go [hunk 5]
```diff
@@ -238,6 +258,25 @@ func resourceClusterExtendParamV3(d *schema.ResourceData, config *Config) map[st
...
+		return []clusters.MasterSpec{}, fmt.Errorf("Error creating HuaweiCloud Cluster: single-master cluster need 1 az for master node, but got %d", len(mastersRaw))
+	}
+	if strings.Split(flavorId, ".")[1] == "s2" && len(mastersRaw) != 3 {
+		return []clusters.MasterSpec{}, fmt.Errorf("Error creating HuaweiCloud Cluster: high-availability cluster need 3 az for master nodes, but got %d", len(mastersRaw))
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_cce_cluster_v3.go]: it's better to return *nil, error* when the validation failed.

File: docs/resources/cce_cluster.md (hunk 1 above)
Comment (Jason-Zhang9309) [reply docs/resources/cce_cluster.md]: fixed

File: docs/resources/cce_cluster.md (hunk 2 above)
Comment (Jason-Zhang9309) [reply docs/resources/cce_cluster.md]: fixed

File: huaweicloud/resource_huaweicloud_cce_cluster_v3.go (hunk 3 above)
Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_cce_cluster_v3.go:133]: fixed

File: huaweicloud/resource_huaweicloud_cce_cluster_v3.go (hunk 4 above)
Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_cce_cluster_v3.go]: fixed

File: huaweicloud/resource_huaweicloud_cce_cluster_v3.go (hunk 5 above)
Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_cce_cluster_v3.go]: fixed

**Pull request merged**